src/ops.rs

```diff
@@ -281,6 +282,23 @@ fn op_code_cache(
   }()))
 }
 
+fn op_chdir(
+  _state: Arc<IsolateState>,
+  base: &msg::Base,
+  data: &'static mut [u8],
+) -> Box<Op> {
+  assert_eq!(data.len(), 0);
+  let inner = base.inner_as_chdir().unwrap();
+  let directory = inner.directory().unwrap();
+  match std::env::set_current_dir(&directory) {
```

Comment:
I think you can just do something like this:
```rust
futures::future::result(std::env::set_current_dir(&directory)
```
I don't think you need the match block at all.

Reply:
added match block to raise error when it could switch to specified directory, will futures:future::result raises? default not found exception?

Reply:
@cedric05 We have a system in place to forward and translate rust errors into deno errors. So the less manipulation of the errors the better. Getting "NotFound" when trying to chdir to a non-existent dir, sounds correct, if that's what you're asking?

It seems this patch is close to landing - you might want to read
[A Shortcut for Propagating Errors: the `?` Operator](https://doc.rust-lang.org/book/second-edition/ch09-02-recoverable-errors-with-result.html#a-shortcut-for-propagating-errors-the--operator)

Feel free to ping me on gitter

Reply:
i will check and try it out tonight.
seems like it will need more changes because of https://github.com/denoland/deno/issues/957